lib/util/http/host-rules.ts

```diff
@@ -162,6 +164,20 @@ export function applyHostRule<GotOptions extends HostRulesGotOptions>(
     options.lookup = dnsLookup;
   }
 
+  if (hostRule.headers) {
+    const allowedHeaders = GlobalConfig.get('allowedHeaders');
+    const filteredHeaders = Object.fromEntries(
+      Object.entries(hostRule.headers).filter(([header]) =>
+        anyMatchRegexOrMinimatch(allowedHeaders, header),
+      ),
```

Comment:
Please rewrite using "for of" loop and if a header does not match then we should log like this:

```
logger.once.warn({ allowedHeaders, header }, 'Disallowed hostRules header');
```